rpcclient/wallet.go

```diff
@@ -758,20 +757,20 @@ type FutureAddMultisigAddressResult chan *response
 // Receive waits for the response promised by the future and returns the
 // multisignature address that requires the specified number of signatures for
 // the provided addresses.
-func (r FutureAddMultisigAddressResult) Receive() (btcutil.Address, error) {
+func (r FutureAddMultisigAddressResult) Receive() (string, error) {
```

Comment:
One option here (and the other places the return type got changed from `btcutil.Adress` to `string`) could be to do 

```golang
type FutureAddMultisigAddressResult  struct {
	response chan *response
	network *chaincfg.Params
}

func (r FutureAddMultisigAddressResult) Receive() (btcutil.Address, error) {
	res, err := receiveFuture(r.response)
	if err != nil {
		return nil, err
	}

	// Unmarshal result as a string.
	var addr string
	err = json.Unmarshal(res, &addr)
	if err != nil {
		return nil, err
	}

	return btcutil.DecodeAddress(addr, r.network)
}

func (c *Client) AddMultisigAddressAsync(requiredSigs int, addresses []btcutil.Address, account string) FutureAddMultisigAddressResult {
	addrs := make([]string, 0, len(addresses))
	for _, addr := range addresses {
		addrs = append(addrs, addr.String())
	}

	cmd := btcjson.NewAddMultisigAddressCmd(requiredSigs, addrs, &account)
	resp :=  c.sendCmd(cmd)
	return FutureAddMultisigAddressResult{network:c.chainParams, response:resp}
}
```

And in

Reply:
Good idea! I think I got all of the address commands with [`327749a`](https://github.com/btcsuite/btcd/pull/1467/commits/327749abd23118bf0daf4e117a7986cdba58b7c6).